app/controllers/api/connect/v3/systems/products_controller.rb

```diff
@@ -5,6 +5,7 @@ class Api::Connect::V3::Systems::ProductsController < Api::Connect::BaseControll
   before_action :check_product_service_and_repositories, only: %i[show activate]
   before_action :load_subscription, only: %i[activate upgrade]
   before_action :check_base_product_dependencies, only: %i[activate upgrade show]
+  after_action :refresh_system_token, only: %i[activate upgrade], if: -> { request.headers.key?(SYSTEM_TOKEN_HEADER) }
```

Comment:
All the `if` conditions for the `refresh_system_token` after_action need a check for `system_tokens_enabled?`, like: 

```suggestion
  after_action :refresh_system_token, only: %i[activate upgrade], if: -> { system_tokens_enabled? && request.headers.key?(SYSTEM_TOKEN_HEADER) }
```

or we include the `system_tokens_enabled?` check into `refresh_system_token()`.

Reply:
it is better to add the check in the refresh_system_token. so added the check.